release-notes/swan-lake-beta4-release-note.md

```diff
@@ -509,6 +509,16 @@ public class CustomRetryManager {
 }
 ```
 
+- A spec deviation was fixed when declaring out of range float value. With current behaviour this gives a compile time error.
+
+- Spec deviation related to numeric literal have been corrected according to the language specification. 
+
+If the numeric literal is not include float type suffix or  decimal type suffix and not a hex floating point literal the type of that numeric literal will be based on following rules.
+
+1. If the literal is floating point literal then the possible basic types in order of preference are float or decimal otherwise they are int, float or decimal.
+2. If there is contextual expected type and there is a basic type int, decimal or float in contextual expected type, Use the most perrered type in the order of int, float or decimal.
+3. Otherwise, use the most preferred possible basic type in the order of int, float or decimal.
```

Comment:
Shall we use the rules from the spec issue. For example, point 2 here doesn't talk about the intersection.

```suggestion
1. If the literal is a floating point literal, then the possible basic types in order of preference are `[float, decimal]`; otherwise they are `[int, float, decimal]`.
2. If there is a contextually expected type C and there is a possible basic type N such that N & C is non-empty, use the most preferred such type.
3. Otherwise, use the most preferred possible basic type.
```